extensions/oplogPopulator/modules/ConnectorsManager.js

```diff
@@ -134,33 +135,29 @@ class ConnectorsManager extends EventEmitter {
         return connector;
     }
 
-    /**
-     * Extracts buckets from a connector config pipeline
-     * @param {Object} connectorConfig connector config
-     * @returns {string[]} list of buckets
-     */
-    _extractBucketsFromConfig(connectorConfig) {
-        const pipeline = connectorConfig.pipeline ?
-            JSON.parse(connectorConfig.pipeline) : null;
-        if (!pipeline || pipeline.length === 0) {
-            return [];
-        }
-        return pipeline[0].$match['ns.coll'].$in;
-    }
-
     /**
      * Gets old connector configs and initializes connector
-     * instances
+     * instances. Deletes connectors as needed if the strategy
+     * rejects them.
      * @param {string[]} connectorNames connector names
      * @returns {Promise|Connector[]} list of connectors
      */
-    async _getOldConnectors(connectorNames) {
+    async _processOldConnectors(connectorNames) {
         try {
             const connectors = await Promise.all(connectorNames.map(async connectorName => {
                 // get old connector config
                 const oldConfig = await this._kafkaConnect.getConnectorConfig(connectorName);
                 // extract buckets from old connector config
-                const buckets = this._extractBucketsFromConfig(oldConfig);
+                const buckets = this._allocationStrategy.getOldConnectorBucketList(oldConfig);
+                if (!buckets) {
+                    await this._kafkaConnect.deleteConnector(connectorName);
+                    this._logger.warn('Removed old connector', {
+                        method: 'ConnectorsManager._processOldConnectors',
```

Comment:
If i understood correctly, when using a single connector to listen to everything, we delete all previous connectors and create a new one with the wildcard element. This might cause us to loose some events or process some events multiple times in case we decide to set a resume time in the config of the connector. Not really sure how we should handle this ?